xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp

```diff
@@ -475,3 +478,274 @@ bool CVaapi1Texture::TestInteropHevc(VADisplay vaDpy, EGLDisplay eglDisplay)
   return ret;
 }
 
+void CVaapi2Texture::Init(InteropInfo& interop)
+{
+  m_interop = interop;
+  m_hasPlaneModifiers = CEGLUtils::HasExtension(m_interop.eglDisplay, "EGL_EXT_image_dma_buf_import_modifiers");
+}
+
+bool CVaapi2Texture::Map(CVaapiRenderPicture* pic)
+{
+#if HAVE_VAEXPORTSURFACHEHANDLE
+  if (m_vaapiPic)
+    return true;
+
+  m_vaapiPic = pic;
+  m_vaapiPic->Acquire();
+
+  VAStatus status;
+
+  VADRMPRIMESurfaceDescriptor surface;
+
+  status = vaExportSurfaceHandle(pic->vadsp, pic->procPic.videoSurface,
+    VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
+    VA_EXPORT_SURFACE_READ_ONLY | VA_EXPORT_SURFACE_SEPARATE_LAYERS,
+    &surface);
+
+  if (status != VA_STATUS_SUCCESS)
+  {
+    CLog::LogFunction(LOGWARNING, "CVaapi2Texture::Map", "vaExportSurfaceHandle failed - Error: %s (%d)", vaErrorStr(status), status);
+    return false;
+  }
+
+  // Remember fds to close them later
+  if (surface.num_objects > m_drmFDs.size())
+    throw std::logic_error("Too many fds returned by vaExportSurfaceHandle");
+
+  for (int object = 0; object < surface.num_objects; object++)
+  {
+    m_drmFDs[object].attach(surface.objects[object].fd);
+  }
+
+  status = vaSyncSurface(pic->vadsp, pic->procPic.videoSurface);
+  if (status != VA_STATUS_SUCCESS)
+  {
+    CLog::LogFunction(LOGERROR, "CVaapi2Texture::Map", "vaSyncSurface - Error: %s (%d)", vaErrorStr(status), status);
+    return false;
+  }
+
+  m_textureSize.Set(pic->DVDPic.iWidth, pic->DVDPic.iHeight);
+
+  GLint type;
+  glGetIntegerv(GL_IMPLEMENTATION_COLOR_READ_TYPE, &type);
+  if (type == GL_UNSIGNED_BYTE)
+    m_bits = 8;
+  else if (type == GL_UNSIGNED_SHORT)
+    m_bits = 16;
+  else
+  {
+    CLog::Log(LOGWARNING, "Did not expect texture type: %d", static_cast<int> (type));
+    m_bits = 8;
+  }
+
+  for (int layerNo = 0; layerNo < surface.num_layers; layerNo++)
+  {
+    int plane = 0;
+    auto const& layer = surface.layers[layerNo];
+    if (layer.num_planes != 1)
+    {
+      CLog::LogFunction(LOGDEBUG, "CVaapi2Texture::Map", "DRM-exported layer has %d planes - only 1 supported", layer.num_planes);
```

Comment:
Can this restriction be removed from Kodi?
Video playback on Intel Rocketlake CPUs is broken atm ( https://github.com/xbmc/xbmc/issues/19820 ).

mpv handles multiple planes just well and has no problems with video playback on Intel Rocketlake CPUs with this code:
https://github.com/mpv-player/mpv/commit/05cb8d28afd8881667c3dea297b182a2ecdd91c8#diff-38c39325f6dad76d975c9eee56dc8a14487c425bae13c3f0482f7fe0fbce2fbbR363

Reply:
Well the API for DRMPRIME external buffer descriptors say that when calling vaExportSurfaceHandle with VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2 and VA_EXPORT_SURFACE_SEPARATE_LAYERS, each layer will contain exactly one plane, so that's why it is handled like this here. If Intel broke this, I guess it's their choice? Considering they're very likely also the ones that originally specified it like this. Anyway, I would be open to adding a quirk if it's not much effort (i.e., it "just works" if we add the other plane properties). PRs welcome.

And please don't necro three year old PRs with "reviews" :-)